aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,486 @@
+  
+The ``Index`` view:
+
+.. literalinclude::   forms/sample/final/Views/Home/Index.cshtml
+  :language: HTML
+  :emphasize-lines: 4
+  
+Which generates the following HTML:
+
+.. code-block:: HTML  
+  :linenos:
+  :emphasize-lines: 2-6 
+
+  <form method="post" action="/">
+    <select id="Country" name="Country">
+      <option value="MX">Mexico</option>
+      <option selected="selected" value="CA">Canada</option>
+      <option value="US">USA</option>
+    </select> 
+      <br /><button type="submit">Register</button>
+    <input name="__RequestVerificationToken" type="hidden" value="<removed for brevity>" />
+  </form>
+
+:Note: We do not recommend using ``ViewBag`` or ``ViewData`` with the Select Tag Helper. A view model is more robust at providing MVC metadata and generally less problematic. 
+
+The ``asp-for`` attribute value doesn't require a ``Model`` prefix, while ``asp-items does``:
+
+.. literalinclude::   forms/sample/final/Views/Home/Index.cshtml
+  :language: HTML
+  :lines: 4
+  :dedent: 3
+
+The ``asp-for`` attribute value is the right hand side of a lambda expression. For example, ``asp-for=""Property1"`` becomes ``m => m.Property1`` in the generated code which is why you don't need to prefix with ``Model``.
+  
+Enum binding
+^^^^^^^^^^^^^^
+
+It's often convenient to use an ``enum`` for a list. The ``CountryEnumViewModel`` uses the ``CountryEnum`` to generate a list of countries:
+
+.. literalinclude::  forms/sample/final/ViewModels/CountryEnumViewModel.cs
+  :language: c#
+  :lines: 3-6
+  :dedent: 3
+  
+.. literalinclude::  forms/sample/final/ViewModels/CountryEnum.cs
+  :language: c#
+  :lines: 1-4,6,8-
+
+The `GetEnumSelectList <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Rendering/IHtmlHelper/index.html>`__ method generates a `SelectList <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Rendering/SelectList/index.html>`__ object for an enum.
+
+.. literalinclude::   forms/sample/final/Views/Home/IndexEnum.cshtml
+  :language: HTML
+  :emphasize-lines: 4-6
+
+You can decorate your model with the ``Display`` attribute to get a richer UI:
+
+.. literalinclude::  forms/sample/final/ViewModels/CountryEnum.cs
+  :language: c#
+  :linenos:
+  :emphasize-lines: 5,7
+
+.. image:: forms/_static/enum.png 
+
+Option Group
+^^^^^^^^^^^^^^^^^^^^^
+
+The HTML  `<optgroup> <https://www.w3.org/wiki/HTML/Elements/optgroup>`__ element is generated when the view model contains one or more `SelectListGroup  <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Rendering/SelectListGroup/index.html>`__ objects.
+
+The ``CountryViewModelGroup`` groups the  ``SelectListItem`` elements into the "North America" and "Europe" groups:
+
+.. literalinclude::  forms/sample/final/ViewModels/CountryViewModelGroup.cs
+  :language: c#
+  :lines: 6-56
+  :dedent: 3
+  :emphasize-lines: 8-9,17,23,35,41,47,29
+
+The two groups are shown below:
+
+.. image:: forms/_static/grp.png  
```

Comment:
Not sure about this image either. If it's included, should be _in addition to_ the generated HTML.